cvat/apps/dataset_manager/annotation.py

```diff
@@ -947,15 +944,15 @@ def interpolate(shape0, shape1):
 
             return shapes
 
-        def propagate(shape, end_frame, *, included_frames=None):
+        def propagate(shape: dict, end_frame, *, included_frames=None):
             return [
                 copy_shape(shape, i)
                 for i in range(shape["frame"] + 1, end_frame)
                 if included_frames is None or i in included_frames
             ]
 
         shapes = []
-        prev_shape = None
+        prev_shape: Optional[dict] = None
```

Comment:
```suggestion
        prev_shape: dict | None = None
```

We're using `|` since it's allowed in 3.10.